mlrun/feature_store/api.py

```diff
@@ -662,6 +662,9 @@ def preview(
     :param sample_size:    num of rows to sample from the dataset (for large datasets)
     """
     # preview reads the source as a pandas df, which is not fully compatible with spark
+    if isinstance(source, pd.DataFrame):
+        source = _rename_source_dataframe_columns(source)
```

Comment:
I guess the operation is idempotent so it's not a big deal, but this will now be called twice on `ingest` (once from `ingest` and once from `preview` that is called by `ingest`).

Reply:
I know, but we have to do it if we want to remain consistent.